src/parser/get_executable_path.c

```diff
@@ -0,0 +1,57 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <libft.h>
+#include <parser/get_executable_path.h>
+
+static void	append_command_to_path(char	*buffer, const char *command)
+{
+	const int	cmd_len = ft_strlen(command) + TERMINATOR;
+
+	*buffer = FORWARD_SLASH;
+	buffer += 1;
+	ft_strlcpy(buffer, command, cmd_len);
+}
+
+static int	get_sub_path_len(const char *path)
+{
+	int	i;
+
+	i = 0;
+	while (path && path[i] != '\0' && path[i] != COLON)
+		i++;
+	return (i);
+}
+
+static void	get_single_path(const char *all_paths, char *buffer, int size)
+{
+	ft_strlcpy(buffer, all_paths, size + TERMINATOR);
+}
+
+int	is_executable(char *full_path_executable, struct stat *status)
+{
+	return (stat(full_path_executable, status));
+}
+
+char	*get_executable_path(const char *command)
+{
+	const char	*all_paths = getenv(PATH);
+	char		*executable_buffer;
+	int			single_path_len;
+	struct stat	status;
+
+	executable_buffer = calloc(CMD_BUFFER_SIZE, sizeof(char));
```

Comment:
CMD_BUFFER_SIZE should be bigger to prevent overflow, I'd assume something like 4096.
Instead of using calloc, on line 38 you can do:
`char exectable_buffer[CMD_BUFFER_SIZE];`
This way the buffer will be in the stack and the memory will be freed automatically at the end of the function scope, so we can have a buffer with a bigger size and we don't need to check if the calloc failed, otherwise, in line 43 for example if !command we still have to free the buffer, same before returning on line 56 before returning, with the array we don't need to worry about it. I know you return the buffer, we can call` ft_strdup(buffer)` in the return statement on line 53, so it will malloc just the memory that is necessary, and return NULL in case of failure.
I think we have to "cleanup" buffer every time we start the while loop, right? So maybe something like this:
`ft_bzero(&executable_buffer[0], CMD_BUFFER_SIZE);`
this syntax `&executable_buffer[0]` can be used in this case instead of casting to `char *`.
for the other functions you can still have the argument as `char *` and then you pass the buffer the same way as in bzero.

Reply:
Believe it or not but but thats how i had it first hehe
I couldn't figure out a way to return the path_string, very nice trick with the str_dup!